### ISSUE TITLE
Extend rules MD051 and MD057 to consider paths in frontmatter

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#772]
### Problem or motivation

Two rules handle link destinations within a set of Markdown:

- Rule MD051: https://github.com/rvben/rumdl/blob/b2ae2baa4893b017bfb985e12298591fe3fcd43c/docs/md051.md
- Rule MD057: https://github.com/rvben/rumdl/blob/b2ae2baa4893b017bfb985e12298591fe3fcd43c/docs/md057.md

In addition, `rumdl` has the ability to detect relative paths in Markdown frontmatter for excluding them from MD044: https://github.com/rvben/rumdl/blob/b2ae2baa4893b017bfb985e12298591fe3fcd43c/docs/md044.md#special-cases.

Given `rumdl` can recognise paths in frontmatter already given specific patterns, it makes sense that rules that deal with file paths would also support these patterns in frontmatter.

For Markdown that stores file paths in frontmatter (such as the scenario given in https://github.com/rvben/rumdl/issues/743), this provides another way of validating correctness.

### Proposed solution

I could see both MD051 and MD057 supporting link path checking in frontmatter.

I could also see `ignore-frontmatter-fields` being added to each rule as optional configuration in case the new functionality created false positives for some.

### Alternatives considered

An alternative would be something like `check-frontmatter-fields` for both MD051 and MD057, where instead of relying on `rumdl`'s existing pattern matching, you can explicitly tell `rumdl` what fields should be considered paths.

To me, though, it makes sense for multiple rules to share logic if possible.

### Feature category

Improve existing rule

### Example markdown

If the feature request was implemented, the path in `link` and the path in the regular Markdown link would be checked:

```markdown
---
title: Heading for MyApp
description: A description for MyApp
link: 'this/is/a/link/to/myapp.md'
---

This is a [link for `myapp`](this/is/a/link/to/myapp.md).
```

**Comment by rvben** [comment]
Shipped in v0.2.48. Both rules take `check-frontmatter`, off by default:

```toml
[MD051]
check-frontmatter = true

[MD057]
check-frontmatter = true
```

On your example MD057 reports the `link:` field when the path does not exist, and MD051 reports a frontmatter fragment that matches no heading in the file. Frontmatter paths are found the same way MD044 finds them, so there is one notion of "this field holds a path" rather than two that can drift.

Off by default, so nothing changes for anyone whose frontmatter holds paths that are not meant to resolve locally.

Thanks @eread, and your instinct about shared logic was the right one: MD044 already knew which frontmatter fields hold paths, so this was a matter of letting the link rules ask.
